numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1345,6 +1345,7 @@ array_richcompare(PyArrayObject *self, PyObject *other, int cmp_op)
         result = PyArray_GenericBinaryFunction(self, other, n_ops.less_equal);
         break;
     case Py_EQ:
+        RICHCMP_GIVE_UP_IF_NEEDED(obj_self, other);
```

Comment:
Geez, I'm not being very bright today! And that only shows that my test is wrong too.  Will fix!